### PR TITLE
Variable cleanup and optimization and clang-cl related fixes

### PR DESCRIPTION
**Pull request by siradam7th**
[WIP] Highlights:

- Made the CMakeLists.txt disable resource files '.rc'  when building with clang-cl.exe (Windows) since they are not supported, and small code changes to avoid compile-time errors: https://github.com/RigsOfRods/rigs-of-rods/commit/8a45d34a177cd472750fa4ed41edcb6292a95083
- Removed, commented or moved some variables to either optimize the code a bit or remove unused variables: https://github.com/RigsOfRods/rigs-of-rods/commit/8c2caa35a478bf9f54cc2961544815fd3545085f https://github.com/RigsOfRods/rigs-of-rods/commit/d9484519ca73d75c996b641b544ae0a042b54c43

### CODE REVIEW

**Comment by ulteq** [comment]
I think we went from using tabs to spaces a while ago.

File: source/main/scripting/ScriptEngine.cpp [hunk 1]
```diff
@@ -425,7 +425,7 @@ int ScriptEngine::framestep(Real dt)
     }
 
     // framestep stuff below
-    if (frameStepFunctionPtr<=0) return 1;
+    if (frameStepFunctionPtr==0) return 1;
```
Comment (only-a-ptr) [inline source/main/scripting/ScriptEngine.cpp]: to check pointers, always use `==nullptr`, not operator ! or 0/NULL comparsion

File: source/main/gameplay/Landusemap.cpp [hunk 2]
```diff
@@ -158,6 +157,7 @@ int Landusemap::loadConfig(Ogre::String filename)
 
         bool bgr = colourMap->getPixelBox().format == PF_A8B8G8R8;
 
+		const Vector3 mapsize = App::GetSimTerrain()->getMaxTerrainSize();
```
Comment (ulteq) [inline source/main/gameplay/Landusemap.cpp]: Tabs -> Spaces

File: source/main/gameplay/Landusemap.cpp [hunk 3]
```diff
@@ -59,8 +59,8 @@ ground_model_t* Landusemap::getGroundModelAt(int x, int z)
 {
     if (!data)
         return 0;
-    Vector3 mapsize = App::GetSimTerrain()->getMaxTerrainSize();
 #ifdef USE_PAGED
+	const Vector3 mapsize = App::GetSimTerrain()->getMaxTerrainSize();
```
Comment (ulteq) [inline source/main/gameplay/Landusemap.cpp]: Tabs -> Spaces

File: source/main/CMakeLists.txt [hunk 4]
```diff
@@ -283,7 +283,10 @@ generate_source_groups(${SOURCE_FILES})
 
 
 if (WIN32)
-    list(APPEND SOURCE_FILES "icon.rc")
+	# clang-cl doesn't support resource files
```
Comment (ulteq) [inline source/main/CMakeLists.txt]: Tabs -> Spaces

File: source/configurator/CMakeLists.txt [hunk 5]
```diff
@@ -9,7 +9,10 @@ project(RoR_Configurator)
 
 if (WIN32)
     set(OS_LIBS "${PThread_LIBRARIES};dinput8.lib;dxguid.lib")
-    set(OS_SOURCE "icon.rc")
+    # clang-cl doesn't support resource files
```
Comment (ulteq) [inline source/configurator/CMakeLists.txt]: Tabs -> Spaces

Comment (siradam7th) [reply source/configurator/CMakeLists.txt]: edited before changing the editor's settings to use spaces,it should be fixed in the next commit

**Comment by siradam7th** [comment]
this should be the end of this pull request unless there is further code changing comments.

**Comment by only-a-ptr** [comment]
🎉 The end!

**Pull request merged**